feature/interface/singleton/otg_tests/singleton_test/README.md

```diff
@@ -73,6 +73,21 @@ a new testbed configuration with the desired port types.
 *   Ensure inbound and outbound unicast counters are the same
 *   Ensure counters increment at the selected SAMPLE interval
 
+### RT-5.1.4 [TODO: https://github.com/openconfig/featureprofiles/issues/2338]
+#### Removal of Interface config should return an error if breakout-mode is still configured
+
+*   On DUT Port-1 with a QSFP-DD 400GBASE-DR4 transceiver inserted
+*   Ensure no breakout is configured
+*   Set Port-1 port-speed to 100GB
+*   The DUT should reject the configuration
```

Comment:
Is this really required?

This requirement implies that explicitly configured port speeds should be validated against the currently equipped transceiver (state) in the mgmt layer.

This coupling of the intended config and current operational state:
1) adds complexity (each commit has to be validated against the current state),
2) limits the ability of an operator to change the intended state (what if this port is supposed to become 100G, and 400G-DR4 optics will be replaced)
3) introduces inconsistencies and corner cases such as
    * what should happen if a port is configured as 100G first, and after that 400G optics is inserted? 

cc @xw-g @dplore @robshakir, thoughts?

Reply:
@LimeHat The problem we are trying to address is that breakout should not be implicitly configured by the DUT If gNMI client sets ```/interfaces/interface/ethernet/config/port-speed``` equal to the breakout speed (e.g. 100G) without actually configuring a breakout using ```/components/component/port/breakout-mode```. A vendor does this right now i.e. if gNMI client sets port-speed of 100G on multilane 400G port, it creates breakouts automatically.
We do not want the breakouts configured automatically. Does Nokia's implementation differ and does Nokia require gNMI client to explicitly set interface and breakout config (if a breakout is needed)?
We do not ask for adding complexity, the operation can error out just like if I try to set port-speed on a 10G interface to 100G, the operation will be illegal and will error-out (couldn't try it yet, but I believe the config will not be accepted)

Reply:
> The problem we are trying to address is that breakout should not be implicitly configured by the DUT If gNMI client sets /interfaces/interface/ethernet/config/port-speed equal to the breakout speed (e.g. 100G) without actually configuring a breakout

That's fair. However, I think that the proposed solution should not be universally required.

>Does Nokia's implementation differ and does Nokia require gNMI client to explicitly set interface and breakout config (if a breakout is needed)?

Our approach is to always require an explicit breakout configuration (we never do it implicitly). Also, we'll reject a breakout config if a port-speed is configured on the interface.

>We do not ask for adding complexity, the operation can error out just like if I try to set port-speed on a 10G interface to 100G, the operation will be illegal and will error-out (couldn't try it yet, but I believe the config will not be accepted)

It may be a simple error condition in some implementations, but I think there are some fundamental aspects to it.   
We don't do enforcement based on the equipped transceiver types for the reasons I listed above, and # 3 is crucial - it means that the same config will be transitioning between valid and invalid states simply based on a transceiver presence (Schrödinger's config?).   

There are certain rules about what port-speeds are accepted, and we can discuss in more detail in buganizer if needed, but our general approach is to let an operator figure out his intent. 
* E.g. If an operator configures 100G speed on a port equipped with 400G transceiver, maybe he wants this port to be up only when a NOC person changes the transceiver in the node, and as soon as he does that, but not earlier. We consider this to be a valid approach. 

Considering all of the above, I can suggest the following:
1) Change the requirement to something like "DUT should not create breakout interfaces implicitly if this configuration is accepted, or, alternatively, the configuration can be rejected"
2) You can also add a requirement that setting port-speed on the interface configured as a breakout should be forbidden and rejected

Reply:
Had a conversation with @dplore and we agreed that
- DUT should not create breakout implicitly if, e.g. on a 400Gig interface, port-speed is configured to be 100G
  - ```/interfaces/interface/ethernet/config/port-speed``` can be configured to 100G by gNMI client but ```/interfaces/interface/ethernet/state/port-speed``` reports the physical port-speed (400G)
  - We have a use case where we need to configure interfaces that do not have SFP's inserted, enforcing validations can cause this use case to break, hence the DUT can accept the config but it may not apply
- Setting port-speed should be rejected on breakout interfaces since these are mutually exclusive configurations